volume-providers/iam/token_exchange.go

```diff
@@ -225,5 +225,15 @@ func (r *tokenExchangeRequest) sendTokenExchangeRequest() (*tokenExchangeRespons
 }
 
 func isConnectionError(err error) bool {
-	return err != nil && strings.Contains(err.Error(), "connection")
+	if err != nil {
```

Comment:
Have you verified with SL in case of connection failure it might be its not returning `tcp` in the error. Please check the SL as well in case of connection error.